fastparquet/test/test_output.py

```diff
@@ -381,15 +381,25 @@ def test_write_compression_schema(tempdir):
 
 
 def test_index(tempdir):
+    import json
     fn = os.path.join(tempdir, 'tmp.parq')
     df = pd.DataFrame({'x': [1, 2, 3],
                        'y': [1., 2., 3.]},
                        index=pd.Index([10, 20, 30], name='z'))
 
     writer.write(fn, df)
 
-    r = ParquetFile(fn)
-    assert set(r.columns) == {'x', 'y', 'z'}
+    pf = ParquetFile(fn)
+    assert set(pf.columns) == {'x', 'y', 'z'}
+    meta = json.loads(pf.key_value_metadata['pandas'])
+    assert meta['index_columns'] == ['z']
```

Comment:
We currently check against duplicate column names after reset_index, so you would get an error for this case